simple_calc/src/components/Form.js

```diff
@@ -0,0 +1,95 @@
+import React from "react"
+import Display from "./display"
+import operation from "./operations"
+
+class Form extends React.Component {
+  state = {
+   numValue: [],
+   operation: "",
+   prevValue: "",
+   display: 0
+
+  };
+
+  convertNum = (num) => {
+    // debugger
+    let newNum = -Math.abs(num);
+  this.setState({
+    numValue: [newNum],
+    display: newNum
+  }) 
+   
+  }
+
+  clearDisplay =() =>{
+    this.setState({
+      display: '0'
+    })
+  }
+
+  handleNumber = (e)=>{
+    e.preventDefault();
+    let btnval = this.state.numValue;
+    btnval += e.target.value
+    this.setState({numValue:[btnval], display:btnval})
+    console.log(this.state.numValue)
+   
+  }
+
+  handleOperation = (e)=>{
+    e.preventDefault();
+    let val = this.state.numValue;
+    let operation = e.target.value
+    this.setState({numValue:[""],prevValue:val, operation:operation})
+    // debugger
+  }
+
+  passingOperationg = (num1,num2,op) => {  
+  let value = operation(num1,num2,op)
+  this.setState({display: value});
+  }
+
+
+  render(){
+    let {numValue, operation, prevValue, display} = this.state
+    return (
+        <div className="calculator">
+       
+        <Display display = {display} />
+        <form onClick ={this.handleNumber} className="calculator-keys">
+            <button value = {0}>0</button>
```

Comment:
You could have had an array of values and looped through them and created a button for each value you see, and attach an event listener accordingly